Argo/Operators/JSONOperators.swift

```diff
@@ -10,39 +10,39 @@ public func <|<A: JSONDecodable>(json: JSONValue, keys: [String]) -> A? {
 }
 
 // Pull value from JSON
-public func <|<A: JSONDecodable>(json: JSONValue, key: String) -> A? {
+public func <|<A where A: JSONDecodable, A == A.DecodedType>(json: JSONValue, key: String) -> A? {
   return json <| [key]
 }
 
 // Pull embedded optional value from JSON
-public func <|?<A: JSONDecodable>(json: JSONValue, keys: [String]) -> A?? {
+public func <|?<A where A: JSONDecodable, A == A.DecodedType>(json: JSONValue, keys: [String]) -> A?? {
   return pure(json <| keys)
 }
 
 // Pull optional value from JSON
-public func <|?<A: JSONDecodable>(json: JSONValue, key: String) -> A?? {
+public func <|?<A where A: JSONDecodable, A == A.DecodedType>(json: JSONValue, key: String) -> A?? {
   return json <|? [key]
 }
 
 // MARK: Arrays
 
 // Pull embedded array from JSON
-public func <||<A: JSONDecodable>(json: JSONValue, keys: [String]) -> [A]? {
+public func <||<A where A: JSONDecodable, A == A.DecodedType>(json: JSONValue, keys: [String]) -> [A]? {
   return json.find(keys) >>- JSONValue.map
 }
 
 // Pull array from JSON
-public func <||<A: JSONDecodable>(json: JSONValue, key: String) -> [A]? {
+public func <||<A where A: JSONDecodable, A.DecodedType == A>(json: JSONValue, key: String) -> [A]? {
```

Comment:
Style: this is inconsistent with every other type constraint. Suggest you change to `A == A.DecodedType` for readability.